acceptance/acceptance_test.go

```diff
@@ -0,0 +1,205 @@
+		// TODO: The ":Z" option is required on Linux systems because of to selinux. This might not
+		// work on a mac, for example.
+		fmt.Sprintf("%s:%s:Z", ts.contextDir, mountedTS.contextDir),
+	}
+
+	cmd := []string{
+		"create",
+		"--store",
+		storePath,
+		fmt.Sprintf("%s=%s", resultFile, sourceFile),
+	}
+
+	if err := runContainer(ctx, cmd, binds); err != nil {
+		return ctx, fmt.Errorf("creating artifact: %w", err)
+	}
+
+	return ctx, nil
+}
+
+func useArtifactForFile(ctx context.Context, fname string) (context.Context, error) {
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return ctx, fmt.Errorf("useArtifactForFile get test state: %w", err)
+	}
+
+	resultInfo, err := readFileContents(filepath.Join(ts.resultsDir(), fname))
+	if err != nil {
+		return ctx, fmt.Errorf("reading result file: %w", err)
+	}
+
+	// Set up the file paths as they will be seen within the container.
+	mountedTS := testState{contextDir: "/data"}
+	storePath := mountedTS.artifactsDir()
+	restoredPath := mountedTS.restoredDir()
+
+	binds := []string{
+		// TODO: The ":Z" option is required on Linux systems because of to selinux. This might not
+		// work on a mac, for example.
+		fmt.Sprintf("%s:%s:Z", ts.contextDir, mountedTS.contextDir),
+	}
+
+	cmd := []string{
+		"use",
+		"--store",
+		storePath,
+		fmt.Sprintf("%s=%s", resultInfo, restoredPath),
+	}
+
+	if err := runContainer(ctx, cmd, binds); err != nil {
+		return ctx, fmt.Errorf("creating artifact: %w", err)
+	}
+
+	return ctx, nil
+}
+
+func restoredFileShouldMatchSource(ctx context.Context, fname string) (context.Context, error) {
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return ctx, fmt.Errorf("useArtifactForFile get test state: %w", err)
+	}
+
+	// To make diffs easier to read, split content by line.
+	toList := cmpopts.AcyclicTransformer("toList", func(in string) []string {
+		return strings.Split(in, "\n")
+	})
+
+	sourceContent, err := readFileContents(filepath.Join(ts.sourceDir(), fname))
+	if err != nil {
+		return ctx, fmt.Errorf("reading source file: %w", err)
+	}
+
+	restoredContent, err := readFileContents(filepath.Join(ts.restoredDir(), fname))
+	if err != nil {
+		return ctx, fmt.Errorf("reading restored file: %w", err)
+	}
+
+	// When comparing for equality, don't "prettify" the file to ensure the restored content is an
+	// identical match to the source. Only use the `toList` transformer when displaying the diff to
+	// help debug issues.
+	if !cmp.Equal(sourceContent, restoredContent) {
+		return ctx, fmt.Errorf("source file does not match restored file: \n%s",
+			cmp.Diff(sourceContent, restoredContent, toList))
+	}
+
+	return ctx, nil
+}
+
+func readFileContents(path string) (string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("opening file %s: %w", path, err)
+	}
+	defer f.Close()
+	content, err := io.ReadAll(f)
+	if err != nil {
+		return "", fmt.Errorf("reading file %s: %w", path, err)
+	}
+	return string(content), nil
```

Comment:
Could be a bit simpler with `os.ReadFile`:
```suggestion
	content, err := os.ReadFile(path)
	return string(content), err
```

Reply:
Done